Sources/SourceKitLSP/Swift/MacroExpansion.swift

```diff
@@ -79,6 +86,10 @@ extension SwiftLanguageService {
 
         let macroExpansionBufferDirectoryURL = self.generatedMacroExpansionsPath
           .appendingPathComponent(macroExpansionBufferDirectoryName)
+        completeExpansionFilePath =
+          completeExpansionFilePath
+          .appendingPathComponent(macroExpansionBufferDirectoryName)
```

Comment:
Doesn’t this add another component for every macro expansion buffer and result in file names like `path/to/temp/macroBuffer1.swift/macroBuffer2.swift/macroBuffer3.swift`. If we want to concatenate them, I think we should concatenate them into a single directory name and not create nested directories.

Also, maybe we should just us a hash of all the macro expansions that can exist. I can imagine that if we concatenate the macro buffer names, we very quickly exceed the 260 character maximum file path length  on Windows. https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

Reply:
I first thought of concatenating the buffer names into a single directory name, and I realised the character limit issue myself.

So, I decided to make sub directories. The sub directories won't have the `.swift` extension and thus the location will be like `path/to/temp/macroBuffer1/macroBuffer2/macroBuffer3/SourceFile.swift`.

I am not sure about the "hash" idea which you are talking about, perhaps you could enlighten me on this.

Reply:
I was thinking that you essentially build up a string `macroBuffer1-macroBuffer2-macroBuffer3` and then, to shorten it, take an MD5 (or some other but MD5 is fine because it doesn’t need to be cryptographic) hash of it, similar to 

https://github.com/swiftlang/sourcekit-lsp/blob/e5d93e1ddd218811693221abe1c45230752d38b6/Sources/LSPLogging/CustomLogStringConvertible.swift#L67

That way we get a directory name of a constant length.